include/enrico/comm.h

```diff
@@ -178,46 +205,110 @@ class Comm {
 };
 
 template<typename T>
-std::enable_if_t<std::is_scalar<std::decay_t<T>>::value> Comm::broadcast(T& value) const
+std::enable_if_t<std::is_scalar<std::decay_t<T>>::value>
+Comm::send_and_recv(T& value, int dest, int source) const
 {
-  this->Bcast(&value, 1, get_mpi_type<T>());
+  if (active() && dest != source) {
+    if (rank == source) {
+      MPI_Send(&value, 1, get_mpi_type<T>(), dest, 81620, comm);
+    } else if (rank == dest) {
+      MPI_Recv(&value, 1, get_mpi_type<T>(), source, MPI_ANY_TAG, comm, MPI_STATUS_IGNORE);
+    }
+  }
+};
+
+template<typename T>
+void Comm::send_and_recv(std::vector<T>& values, int dest, int source) const
+{
+  if (active() && dest != source) {
+    // Send the size of the vector from the source
+    auto n = values.size();
+    send_and_recv(n, dest, source);
+    // Resize the vector on dest
+    if (rank == dest && values.size() != n) {
+      values.resize(n);
+    }
+
+    // Send the vector
+    int tag = source;
```

Comment:
Sure, would you rather use `source` for the tag on both the send and recv calls?  I don't consider it to be much more robust or more necessary than using `81620` and `MPI_ANY_TAG`.  We're still relying on **a lot** of synchronous behavior from whatever ranks are calling `send_and_recv`, so I don't think tag matching contributes any functionality.

Reply:
Right, in practice it won't matter that much. Either way is fine with me. I guess I would lean toward having matching tags for the sender/received since it doesn't hurt

Reply:
done!